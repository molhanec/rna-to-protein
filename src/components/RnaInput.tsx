import React, { useState } from 'react'
import { FormGroup, Label, InputGroup, Input, InputGroupAddon, Button } from 'reactstrap'

interface Props {
  onTranslate: (rna: string) => void
  invalid: boolean
}

const rnaExample = 'AUGGCCAUGGCGCCCAGAACUGAGAUCAAUAGUACCCGUAUUAACGGGUGA'

const RnaInput = ({ onTranslate, invalid }: Props) => {

  const [rna, setRna] = useState(rnaExample)

  return (
    <FormGroup>
      <Label>
        RNA codons
      </Label>
      <InputGroup>
        <Input
          value={rna}
          invalid={invalid}
          onChange={event => setRna(event.target.value)}
          onKeyPress={event => event.key === 'Enter' && onTranslate(rna)}
        />
        <InputGroupAddon addonType="append">
          <Button
            color="primary"
            disabled={rna.trim() === ''}
            onClick={() => onTranslate(rna)}
          >
            Translate
          </Button>
          <Button
            color="secondary"
            onClick={() => {
              setRna(rnaExample)
              onTranslate(rnaExample)
            }}
          >
            Reset to example
          </Button>
        </InputGroupAddon>
      </InputGroup>
    </FormGroup>
  )
}

export default RnaInput
