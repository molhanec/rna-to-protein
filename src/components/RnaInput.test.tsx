import React from 'react'
import { render } from '@testing-library/react'
import RnaInput from './RnaInput'

describe('<RnaInput />', () => {
  it('matches snapshot', () => {
    const { container } = render(<RnaInput rna="RNA" setRna={() => { }} />)
    expect(container).toMatchSnapshot()
  })
})
