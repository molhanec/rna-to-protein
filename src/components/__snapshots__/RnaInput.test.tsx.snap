// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RnaInput /> matches snapshot 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=""
    >
      RNA codons
    </label>
    <div
      class="input-group"
    >
      <input
        class="is-invalid form-control"
        type="text"
        value="AUGGCCAUGGCGCCCAGAACUGAGAUCAAUAGUACCCGUAUUAACGGGUGA"
      />
      <div
        class="input-group-append"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Translate
        </button>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Reset to example
        </button>
      </div>
    </div>
  </div>
</div>
`;
